### PR TITLE
Removed UniquePTR from PathFinder

### PR DESCRIPTION
**Pull request by SafwatHalaby**
@Woazboat noticed it's not needed, so I removed it.

### CODE REVIEW

File: src/Mobs/Path.h [hunk 1]
```diff
@@ -24,7 +24,15 @@ class cChunk;
...
-struct cPathCell;  // Defined inside Path.cpp
+enum class eCellStatus {OPENLIST,  CLOSEDLIST,  NOLIST};
+struct cPathCell
+{
+	Vector3i m_Location;   // Location of the cell in the world.
+	int m_F, m_G, m_H;  // F, G, H as defined in regular A*.
+	eCellStatus m_Status;  // Which list is the cell in? Either non, open, or closed.
+	cPathCell * m_Parent;  // Cell's parent, as defined in regular A*.
+	bool m_IsSolid;	   // Is the cell an air or a solid? Partial solids are currently considered solids.
+};
```
Comment (NiLSPACE) [inline src/Mobs/Path.h]: I think we want 5 empty spaces after this

Comment (SphinxC0re) [inline src/Mobs/Path.h]: You mean lines, don't you?

Comment (NiLSPACE) [inline src/Mobs/Path.h]: Yes, sorry ;)

Comment (SafwatHalaby) [inline src/Mobs/Path.h]: Fixed.

File: src/Mobs/Path.cpp [hunk 2]
```diff
@@ -388,23 +381,20 @@ void cPath::ProcessCell(cPathCell * a_Cell, cPathCell * a_Caller, int a_GDelta)
...
 				si::setBlock(a_Location.x, a_Location.y, a_Location.z, debug_unchecked, Cell->m_IsSolid ? NORMAL : MINI);
 			#endif
 		#endif
-		return Cell;
+		return &m_Map[a_Location];
 	}
 	else
 	{
-		return m_Map[a_Location].get();
+		return &m_Map[a_Location];
```
Comment (SphinxC0re) [inline src/Mobs/Path.cpp]: What does the else do here? you can just return  &m_Map[a_Location]; after closing the if

Comment (SafwatHalaby) [inline src/Mobs/Path.cpp]: True, it's logically pointless and it's just there for code clarity.
I wonder if compilers optimize it away...

Comment (SphinxC0re) [inline src/Mobs/Path.cpp]: I don't think so. Compilers aren't that intelligent

Comment (worktycho) [inline src/Mobs/Path.cpp]: They will, its just basic hoisting.

Comment (SphinxC0re) [inline src/Mobs/Path.cpp]: Okay then I think it's ready for merge

**Comment by SafwatHalaby** [comment]
You should have let me squash it first.

**Pull request merged**